src/main/java/org/cryptomator/ui/mainwindow/VaultListController.java

```diff
@@ -214,9 +256,39 @@ public void didClickAddExistingVault() {
 		addVaultWizard.build().showAddExistingVaultWizard(resourceBundle);
 	}
 
+	@FXML
+	public void didClickRecoverExistingVault() {
+		Optional<File> selectedDirectory = chooseValidVaultDirectory();
+		if (selectedDirectory.isEmpty()) {
+			return;
+		}
+
+		Vault preparedVault = VaultPreparator.prepareVault(selectedDirectory.get(), vaultComponentFactory, mountServices);
+
+		Optional<Vault> matchingVaultListEntry = vaultListManager.get(preparedVault.getPath());
+		if (matchingVaultListEntry.isPresent()) {
+			dialogs.prepareRecoveryVaultAlreadyExists(mainWindow, matchingVaultListEntry.get().getDisplayName()) //
+					.setOkAction(Stage::close) //
+					.build().showAndWait();
+			return;
+		}
```

Comment:
If we compare the vault by path, can't we first check for the path and only if there is no matching entry create/prepare it?